src/participant/participant.ts

```diff
@@ -811,50 +784,205 @@ export default class ParticipantController {
+  }): Promise<string | undefined> {
+    const collections = await this._getCollections({ stream, databaseName });
+
+    if (collections === undefined) {
+      return;
+    }
+    
+    if (collections.length === 1) {
+      return collections[0].name;
+    }
+
+    stream.markdown(
+      vscode.l10n.t(
+        `Which collection would you like to use within ${databaseName}?\n\n`
+      )
+    );
+
+    this.renderCollectionsTree({
+      collections,
+      command,
+      databaseName,
+      context,
+      stream,
+    });
+  }
+
+  /** Gets the database name if there is only one collection.
+   *  Otherwise returns undefined and asks the user to select the database. */
+  async _getOrAskForDatabaseName({
+    command,
+    context,
+    stream,
+  }: {
+    command: ParticipantCommand;
+    context: vscode.ChatContext;
+    stream: vscode.ChatResponseStream;
+  }): Promise<string | undefined> {
+    const databases = await this._getDatabases({ stream });
+
+    if (databases === undefined || databases.length === 0) {
+      log.error('No databases found');
+      return;
+    }
+
+    if (databases.length === 1) {
+      return databases[0].name;
+    }
+
     // If no database or collection name is found in the user prompt,
     // we retrieve the available namespaces from the current connection.
     // Users can then select a value by clicking on an item in the list.
+    stream.markdown(
+      `What is the name of the database you would like${
+        command === '/query' ? ' this query' : ''
+      } to run against?\n\n`
+    );
+
+    this.renderDatabasesTree({
+      databases,
+      command,
+      context,
+      stream,
+    });
+  }
+
+  /** Helper which either automatically picks and returns missing parts of the namespace (if any)
+   *  or prompts the user to pick the missing namespace.
+   */
+  async _getOrAskForMissingNamespace({
+    databaseName,
+    collectionName,
+    context,
+    stream,
+    command,
+  }: {
+    databaseName: string | undefined;
+    collectionName: string | undefined;
+    context: vscode.ChatContext;
+    stream: vscode.ChatResponseStream;
+    command: ParticipantCommand;
+  }): Promise<{
+    databaseName: string | undefined;
+    collectionName: string | undefined;
+  }> {
     if (!databaseName) {
-      stream.markdown(
-        `What is the name of the database you would like${
-          command === '/query' ? ' this query' : ''
-        } to run against?\n\n`
-      );
-      await this.renderDatabasesTree({
+      databaseName = await this._getOrAskForDatabaseName({
         command,
         context,
         stream,
       });
-    } else if (!collectionName) {
-      stream.markdown(
-        `Which collection would you like to use within ${databaseName}?\n\n`
+
+      // If the database name could not get automatically selected,
+      // then the user has been prompted for it instead.
+      if (!databaseName) {
+        return { databaseName, collectionName };
+      }
```

Comment:
I'm not sure I understand how this works. As far as I can follow this, if I don't have any databases, I'd get the "No databases found" error and no prompt to create/select a new one.

Reply:
The comment was not describing everything so I updated that. databaseName ends up as undefined either because we don't have it set yet (so the user has been prompted for it) or if there's an error. It's handled by the `_getOrAskFor...` function.